sfge/src/main/java/com/salesforce/apex/jorje/JorjeUtil.java

```diff
@@ -69,17 +67,30 @@ public static AstNodeWrapper<?> compileApexFromString(String sourceCode) {
         final CodeUnit codeUnit = codeUnits.get(0);
 
         // Find any ParseExceptions. Other exceptions are ignored because the compiler does not have
-        // all of the relevant
-        // pieces needed to check semantics.
+        // all of the relevant pieces needed to check semantics.
+        // Sometimes the same ParseException appears multiple times, so we should use a Set to
+        // make sure that the messages are unique.
+        final Set<String> exceptionMessages = new HashSet<>();
         final List<CompilationException> exceptions =
                 codeUnit.getErrors().get().stream()
-                        .filter(e -> e instanceof ParseException)
+                        // Set.add() returns true if the thing being added wasn't already in the
+                        // set.
+                        .filter(
+                                e ->
+                                        e instanceof ParseException
+                                                && exceptionMessages.add(e.getMessage()))
                         .collect(Collectors.toList());
         if (!exceptions.isEmpty()) {
             throw new JorjeCompilationException(
                     exceptions.stream()
-                            .map(Throwable::getMessage)
-                            .collect(Collectors.joining(",")));
+                            .map(
+                                    e ->
+                                            String.format(
+                                                    "ParseException at %d:%d. (%s)",
```

Comment:
Could you move this to UserFacingMessages.java so that we have all the message constants in the same place? Doc review would be easier.

Reply:
Instead of saying, "ParseException", can we make this read something like, "Invalid syntax at <blah>"? Line, column and error information looks good.